### DISCUSSION TITLE
Firmware and Hardware Differences Between Version

### DISCUSSION THREAD

**Discussion by obsidience** [4IceG/RM520N-GL#11] [Q&A]
Hi all,

What's the difference between R01 and R03?  

My device returns the following firmware version "RM520NGLAAR01A07M4G" and I assume that's an R01 version.  Does this mean that I need to update it to the latest R01 or can it move to the R03 firmware?

Thanks for your help!

**Comment by 4IceG** [comment]
Hi, 
I know someone asked about this on the Quectel forum, but the question remained unanswered. I don't know what exactly is difference between these revisions.

**Reply by obsidience** [reply]
Yeah, it's a bit nebulous.  I read elsewhere, I believe on the gl-inet forums, that it might be a hardware revision.  If that's the case, it's probably best to stick with the revision you started with...  Though some on that forum appear to have started with R01 and were definitely testing R03 - but also reporting buginess maintaining a 5G connection.  I'm sticking with R01 and will close this question.

Thanks!

**Reply by MickMorley** [reply]
There's a couple reports of R01 being more stable than R03 at https://github.com/iamromulan/quectel-rgmii-configuration-notes/issues/8 and https://github.com/iamromulan/quectel-rgmii-configuration-notes/issues/1. Specifically RM520NGLAAR01A07M4G

**Reply by obsidience** [reply]
Thanks for the links; I've read through them.  I think part of the problem might be that some people are trying to mix and match specific firmware to their hardware, which might cause problems.  It's also hard to diagnose what's going on as we don't know specifically what firmware and revision they started from.  Usually, a firmware update takes into account the hardware revision and will not allow a flash but it's also likely that there's less safeguards as Quectel expects manufacturers and providers to be doing this work.
